pytest_xvfb.py

```diff
@@ -49,19 +70,35 @@ def start(self):
         ret = self._proc.poll()
 
         if ret is not None:
+            self._clear_xauthority()
             raise XvfbExitedError("Xvfb exited with exit code {0}".format(ret))
 
     def stop(self):
-        if self._old_display is None:
-            del os.environ['DISPLAY']
-        else:
-            os.environ['DISPLAY'] == self._old_display
+        self._clear_xauthority()
+        self._restore_env()
         try:
             self._proc.terminate()
             self._proc.wait()
         except OSError:
             pass
 
+    def _save_env(self):
+        self._old_env = {}
+        for varname in self.RESTORED_ENVVARS:
+            self._old_env[varname] = os.getenv(varname)
```

Comment:
Are you using `os.getenv` here to get the`None` fallback if it's not set? If so, I'd prefer `os.environ.get(varname)` (i.e. using `dict.get`) as I had to look up what `os.getenv` does with an unknown key :wink: